pkg/reconciler/openshift/tektonconfig/rbac.go

```diff
@@ -399,9 +402,118 @@ func (r *rbac) createRoleBinding(ctx context.Context, sa *corev1.ServiceAccount)
 		Subjects: []rbacv1.Subject{{Kind: rbacv1.ServiceAccountKind, Name: sa.Name, Namespace: sa.Namespace}},
 	}
 
-	_, err = rbacClient.RoleBindings(sa.Namespace).Create(ctx, rb, metav1.CreateOptions{})
-	if err != nil {
+	if _, err := rbacClient.RoleBindings(sa.Namespace).Create(ctx, rb, metav1.CreateOptions{}); err != nil {
 		logger.Error(err, "creation of 'edit' rolebinding failed, in Namespace", sa.GetNamespace())
+		return err
+	}
+	return nil
+}
+
+func (r *rbac) ensureClusterRoleBindings(ctx context.Context, sa *corev1.ServiceAccount) error {
+	logger := logging.FromContext(ctx)
+
+	rbacClient := r.kubeClientSet.RbacV1()
+	logger.Info("finding cluster-role clusterinterceptors-clusterrole")
+	if _, err := rbacClient.ClusterRoles().Get(ctx, "clusterinterceptors-clusterrole", metav1.GetOptions{}); errors.IsNotFound(err) {
+		return r.createClusterRole(ctx)
+	}
+
+	logger.Info("finding cluster-role-binding clusterinterceptors-clusterrolebinding")
+
+	viewCRB, err := rbacClient.ClusterRoleBindings().Get(ctx, "clusterinterceptors-clusterrolebinding", metav1.GetOptions{})
+
+	if err == nil {
+		logger.Infof("found clusterrolebinding %s", viewCRB.Name)
+		return r.updateClusterRoleBinding(ctx, viewCRB, sa)
+	}
+
+	if errors.IsNotFound(err) {
+		return r.createClusterRoleBinding(ctx, sa)
 	}
+
 	return err
 }
+
+func (r *rbac) updateClusterRoleBinding(ctx context.Context, rb *rbacv1.ClusterRoleBinding, sa *corev1.ServiceAccount) error {
+	logger := logging.FromContext(ctx)
+
+	subject := rbacv1.Subject{Kind: rbacv1.ServiceAccountKind, Name: sa.Name, Namespace: sa.Namespace}
+
+	hasSubject := hasSubject(rb.Subjects, subject)
+	if !hasSubject {
+		rb.Subjects = append(rb.Subjects, subject)
+	}
+
+	ownerRef := rb.GetOwnerReferences()
+	if len(ownerRef) == 0 {
+		rb.SetOwnerReferences([]metav1.OwnerReference{r.ownerRef})
+	}
```

Comment:
hi, if this is not `== 0` ideally we should, append our owner reference to existing ones and then set the updated list.
or we should reset it and set ours, as this is a clusterrole created and managed by us.

I think we should not skip setting ownerRefernce under any circumstances.

i don't mind merging this pr as it is.
However, could you create an issue, mentioning this.

1. if ownerReference already exists, check whether one of the ownerReference (kind and name) is ours
2. else append our owner reference.

Reply:
@nikhil-thomas created issue https://github.com/tektoncd/operator/issues/329 to track this

Also i have cross verified that `ownerReference` added properly